AmplifyPlugins/API/AWSAPICategoryPlugin/SubscriptionFactory/AWSSubscriptionConnectionFactory.swift

```diff
@@ -83,7 +83,15 @@ class AWSSubscriptionConnectionFactory: SubscriptionConnectionFactory {
                                                     "When instantiating AWSAPIPlugin pass in an instance of APIAuthProvider",
                                                     nil)
             }
-            let wrappedProvider = OIDCAuthProviderWrapper(oidcAuthProvider: oidcAuthProvider)
+            let wrappedProvider = OIDCAuthProviderWrapper(authTokenProvider: oidcAuthProvider)
+            authInterceptor = OIDCAuthInterceptor(wrappedProvider)
```

Comment:
Please review both `OIDCAuthProviderWrapper` and `OIDCAuthInterceptor`. If either has OIDC-specific logic, consider a new type for the Function custom authorizer. If neither have specific logic, consider renaming to make it a little more generic (like you did with AuthTokenProvider)

Reply:
`OIDCAuthProviderWrapper` [doesn't have any OIDC-specific logic ](https://github.com/aws-amplify/amplify-ios/blob/main/AmplifyPlugins/API/AWSAPICategoryPlugin/Interceptor/SubscriptionInterceptor/OIDCAuthProviderWrapper.swift), I'll rename it to a more generic `AuthTokenProviderWrapper`.

`OIDCAuthInterceptor` is fairly generic as well, but it's a public type coming from the `AppSyncRealTimeClient` package.
 @palpatim we can decider to either continue using the type as is, or "fork" it in Amplify to a new "TokenAuthInterceptor"/"FunctionAuthInterceptor" or similar.

Reply:
If we decided to create a new `TokenAuthInterceptor`/`FunctionAuthInterceptor` we might not need to rename `OIDCAuthProviderWrapper` at all.

Reply:
Added a new `AuthenticationTokenAuthInterceptor`